core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -62,28 +63,30 @@
     /**
      * Creates a new instance.
      */
-    ServiceConfig(Route route, HttpService service,
-                  @Nullable String defaultServiceName, @Nullable String defaultLogName,
-                  long requestTimeoutMillis, long maxRequestLength, boolean verboseResponses,
-                  AccessLogWriter accessLogWriter, boolean shutdownAccessLogWriterOnStop) {
-        this(null, route, service, defaultServiceName, defaultLogName, requestTimeoutMillis, maxRequestLength,
-             verboseResponses, accessLogWriter, shutdownAccessLogWriterOnStop,
-             extractTransientServiceOptions(service));
+    ServiceConfig(Route route, HttpService service, @Nullable String defaultLogName,
+                  ServiceNaming defaultServiceNaming, long requestTimeoutMillis, long maxRequestLength,
+                  boolean verboseResponses, AccessLogWriter accessLogWriter,
+                  boolean shutdownAccessLogWriterOnStop) {
+        this(null, route, service, defaultLogName, defaultServiceNaming,
+             requestTimeoutMillis, maxRequestLength, verboseResponses, accessLogWriter,
+             shutdownAccessLogWriterOnStop, extractTransientServiceOptions(service));
     }
 
     /**
      * Creates a new instance.
      */
     private ServiceConfig(@Nullable VirtualHost virtualHost, Route route, HttpService service,
-                          @Nullable String defaultServiceName, @Nullable String defaultLogName,
-                          long requestTimeoutMillis, long maxRequestLength, boolean verboseResponses,
-                          AccessLogWriter accessLogWriter, boolean shutdownAccessLogWriterOnStop,
+                          @Nullable String defaultLogName, ServiceNaming defaultServiceNaming,
+                          long requestTimeoutMillis, long maxRequestLength,
+                          boolean verboseResponses, AccessLogWriter accessLogWriter,
+                          boolean shutdownAccessLogWriterOnStop,
                           Set<TransientServiceOption> transientServiceOptions) {
         this.virtualHost = virtualHost;
         this.route = requireNonNull(route, "route");
         this.service = requireNonNull(service, "service");
-        this.defaultServiceName = defaultServiceName;
         this.defaultLogName = defaultLogName;
+        this.defaultServiceNaming = defaultServiceNaming;
+        this.defaultServiceName = defaultServiceNaming.convert(this);
```

Comment:
Could we add a quick `null` check for the return value, just in case a user returned `null` by mistake?